dual_net.py

```diff
@@ -0,0 +1,155 @@
+            policy_out = layers.fully_connected(policy_out,
+                                                num_outputs=specs['num_outputs'],
+                                                activation_fn=specs['activation_fn'])
+    policy_out = tf.nn.log_softmax(policy_out)
+
+    # Value head
+    value_out = out
+    for specs in value_head:
+        if specs['layer'] == 'conv':
+            value_out = conv_block(value_out, specs)
+        elif specs['layer'] == 'fc':
+            value_out = layers.flatten(value_out)
+            value_out = layers.fully_connected(value_out,
+                                               num_outputs=specs['num_outputs'],
+                                               activation_fn=specs['activation_fn'])
+    return policy_out, value_out
+
+
+class DualNet(object):
+
+    def __init__(self, sess, action_rep='pos', learning_rate=0.01, regularization_mult=0.01):
+        """
+        sess: tensorflow session
+        action_rep: 'pos' for (pos1, pos2) action representation, 'piece' for (piece, pos) rep
+        learning_rate: learning rate for gradient descent
+        regularization_mult: multiplier for weight regularization loss
+        """
+        self.board_placeholder = tf.placeholder(tf.float32, [None, 8, 8, 13])
+        shared_layers = [{'layer': 'conv', 'num_outputs': 256, 'stride': 3, 'kernel_size': 1, 'activation_fn': tf.nn.relu},
+                         {'layer': 'residual', 'num_outputs': 256, 'stride': 1, 'kernel_size': 3, 'activation_fn': tf.nn.relu},
+                         {'layer': 'residual', 'num_outputs': 256, 'stride': 1, 'kernel_size': 3, 'activation_fn': tf.nn.relu}]
+        if action_rep == 'pos':
+            policy_layers = [{'layer': 'conv', 'num_outputs': 2, 'stride': 1, 'kernel_size': 1, 'activation_fn': tf.nn.relu},
+                             {'layer': 'fc', 'num_outputs': 64*64, 'activation_fn': None}]
+        elif action_rep == 'piece':
+            policy_layers = [{'layer': 'conv', 'num_outputs': 2, 'stride': 1, 'kernel_size': 1, 'activation_fn': tf.nn.relu},
+                             {'layer': 'fc', 'num_outputs': 32*64, 'activation_fn': None}]
+        value_layers = [{'layer': 'conv', 'num_outputs': 1, 'stride': 1, 'kernel_size': 1, 'activation_fn': tf.nn.relu},
+                        {'layer': 'fc', 'num_outputs': 256, 'activation_fn': tf.nn.relu},
+                        {'layer': 'fc', 'num_outputs': 1, 'activation_fn': tf.nn.tanh}]
+
+        self.policy_predict, self.value_predict = build_mlp(self.board_placeholder,
+                                                            scope='net',
+                                                            shared_layers=shared_layers,
+                                                            policy_head=policy_layers,
+                                                            value_head=value_layers)
+        self.z = tf.placeholder(tf.float32, [None])
```

Comment:
If we're going to call the prediction `value_predict` maybe we can call this `value_observed`?